ttnn/cpp/ttnn/operations/data_movement/sharded/device/kernels/dataflow/writer_unary_stick_layout_sharded_blocks_interleaved_start_id.cpp

```diff
@@ -34,6 +40,12 @@ void kernel_main() {
     uint32_t stick_id = start_id;
     cb_wait_front(cb_id_out0, block_height);
     uint32_t l1_read_addr = get_read_ptr(cb_id_out0);
+
+#ifdef DEBUG
+    noc_async_read_barrier();
+    tt::data_movement::common::print_pages(l1_read_addr, block_width_bytes >> 1, block_height);
+#endif
+
     for (uint32_t h = 0; h < block_height; ++h) {
         uint64_t dst_noc_addr = get_noc_addr(stick_id, s0);
         noc_async_write(l1_read_addr, dst_noc_addr, block_width_bytes);
```

Comment:
s2i does not need to handle the case for aligned and not aligned as i2s?
@tt-aho please correct if I'm wrong, but I thought for block/width sharding, some cores need to be aware if the current write location in dram is 32B aliged.

Reply:
DRAM write alignment is the same as L1 alignment on current arches (16B), so alignment is more of an issue for readers rather than writers

Reply:
ah i see, thanks